library/common/engine.cc

```diff
@@ -93,6 +93,19 @@ Engine::~Engine() {
   main_thread_.join();
 }
 
+void Engine::recordCounter(std::string elements, uint64_t count) {
+  if (server_) {
+    server_->dispatcher().post([this, elements, count]() -> void {
+      std::string client = "client";
+      absl::string_view prefix{client};
+      absl::string_view dynamic_elements{elements};
+      Stats::Utility::counterFromElements(server_->serverFactoryContext().scope(),
```

Comment:
Based on my reading of the linked doc, the main cost associated with dynamic tokens is that they take up more space and take time to encode - but they don't actually contend locks. It also appears new static tokens can be created via XDS updates, implying this could be an option for us as well as a potential optimization.

At present, all our work is on the main thread anyhow, and with dispatched stat updates, they're going to execute exclusively between I/O events regardless. I'd be fine with prominently documenting that this interface is experimental and may have unknown performance implications, but I feel we shouldn't block on that any more than we have other potential performance sinks in our hot path (extra copies, etc.).

What do you think?

Reply:
Chatted with @junr03, we're going to flag the StatsClient interface as experimental, and move forward, while continuing to investigate potential performance ramifications.